lib/node_modules/@stdlib/repl/lib/main.js

```diff
@@ -75,6 +86,22 @@ var SETTINGS_VALIDATORS = require( './settings_validators.js' );
 // VARIABLES //
 
 var debug = logger( 'repl' );
+var mergeOptions = mergeFcn({
+	'level': 2 // to merge individual settings and themes
+});
```

Comment:
I am still skeptical if this is the best way to do this. For context, after resolving the options from the configuration, I am merging it with the `defaults()`, till level 2. My main concern being, we might not always want to merge all things till level 2 (and things like the input and output stream properties might also get wrongly merged creating corrupted data), should we have a custom merger (similar to the `validate` function)?

Reply:
the `defaults()` also has `opts.input` which is a stream object (with many nested properties at diff levels), say someone makes a custom configuration, with an `input` option to a different object, they can potentially inject and corrupt the stream object at some levels because of the merge. We do have a list of `PERSISTABLE_OPTIONS` (L94), that we are using to allow saving only those specific properties. Maybe we can use it to also enforce that **only the `PERSISTABLE_OPTIONS` can be loaded through a custom configuration**. But again, this isn't future-proof, as it can be something else other than streams as well that we might later support.

Currently, in the `validate` function, we manually check if the given options (at REPL instantiation) has the prop or not (for a property, say `settings`), if yes, we *merge* or *replace* the default with the given property manually for each option. So, should we do something similar to merge 1) defaults, 2) configuration, and 3) REPL options into a final `opts` object?